symengine/lib/symengine_wrapper.pyx

```diff
@@ -4789,5 +4806,17 @@ def solve(f, sym, domain=None):
     return c2py(<RCP[const symengine.Basic]>(symengine.solve(f_.thisptr, x, d)))
 
 
+def cse(exprs):
+    cdef symengine.vec_basic vec
+    cdef symengine.vec_pair replacements
+    cdef symengine.vec_basic reduced_exprs
+    cdef Basic b
+    for expr in exprs:
+        b = sympify(expr)
+        vec.push_back(b.thisptr)
+    symengine.cse(replacements, reduced_exprs, vec)
+    return (vec_pair_to_list(replacements), vec_basic_to_list(reduced_exprs))
```

Comment:
Are we sure on returning the results in a `tuple`? `SymPy`, if I'm not wrong, doesn't do that and might lead to an inconsistency later.

Reply:
SymPy does return a tuple. (At least in git master)